templates/default/upstart.conf.erb

```diff
@@ -0,0 +1,52 @@
+# <%= @name %> generated by poise-service for <%= @new_resource.to_s %>
+
+description "<%= @name %>"
+
+start on runlevel [2345]
+stop on runlevel [!2345]
+
+<% limits = @new_resource.enclosing_provider.new_resource.limits %>
+<%= "limit nofile #{limits['open_files']} #{limits['open_files']}" %>
```

Comment:
Looks like the nproc stanza is honored:

{code}
[root@default-centos-67 init]# start my_tomcat_2
my_tomcat_2 start/running, process 4526
[root@default-centos-67 init]# cat /proc/4526/limits
Limit                     Soft Limit           Hard Limit           Units     
Max cpu time              unlimited            unlimited            seconds   
Max file size             unlimited            unlimited            bytes     
Max data size             unlimited            unlimited            bytes     
Max stack size            10485760             unlimited            bytes     
Max core file size        0                    unlimited            bytes     
Max resident set          unlimited            unlimited            bytes     
Max processes             4096                 4096                 processes 
Max open files            65536                65536                files     
Max locked memory         65536                65536                bytes     
Max address space         unlimited            unlimited            bytes     
Max file locks            unlimited            unlimited            locks     
Max pending signals       1868                 1868                 signals   
Max msgqueue size         819200               819200               bytes     
Max nice priority         0                    0                    
Max realtime priority     0                    0                    
Max realtime timeout      unlimited            unlimited            us        
[root@default-centos-67 init]# sed -i -- 's/limit nproc 4096 4096/limit nproc 8192 8192/g' my_tomcat_2.conf
[root@default-centos-67 init]# stop my_tomcat_2
my_tomcat_2 stop/waiting
[root@default-centos-67 init]# start my_tomcat_2
my_tomcat_2 start/running, process 4567
[root@default-centos-67 init]# cat /proc/4567/limits
Limit                     Soft Limit           Hard Limit           Units     
Max cpu time              unlimited            unlimited            seconds   
Max file size             unlimited            unlimited            bytes     
Max data size             unlimited            unlimited            bytes     
Max stack size            10485760             unlimited            bytes     
Max core file size        0                    unlimited            bytes     
Max resident set          unlimited            unlimited            bytes     
Max processes             8192                 8192                 processes 
Max open files            65536                65536                files     
Max locked memory         65536                65536                bytes     
Max address space         unlimited            unlimited            bytes     
Max file locks            unlimited            unlimited            locks     
Max pending signals       1868                 1868                 signals   
Max msgqueue size         819200               819200               bytes     
Max nice priority         0                    0                    
Max realtime priority     0                    0                    
Max realtime timeout      unlimited            unlimited            us
{code}